alpha/alpha-server/src/main/java/org/apache/servicecomb/saga/alpha/server/AlphaConfig.java

```diff
@@ -42,37 +45,47 @@
   @Value("${alpha.compensation.retry.delay:3000}")
   private int delay;
 
-  // TODO: 2017/12/27 to be filled with actual callbacks on completion of SCB-138
+  // TODO: 2018/01/03 optimize reverse visit of the map instead of using another map, namely omegaCallbacksReverse
   @Bean
   Map<String, Map<String, OmegaCallback>> omegaCallbacks() {
     return new ConcurrentHashMap<>();
   }
 
+  @Bean
+  Map<StreamObserver<GrpcCompensateCommand>, Pair<String, String>> omegaCallbacksReverse() {
```

Comment:
we use a shared omega callbacks map, in order to use it in service/instance aware callback. why do we need a shared map here?